empirical-data/common/src/main/scala/com/azavea/ingest/common/HydrateRDD.scala

```diff
@@ -62,15 +62,20 @@ object HydrateRDD {
     builder.buildFeature(orig.getID)
   }
 
-  def normalizeShpRdd(rdd: RDD[SimpleFeature], typeName: String) = {
-    val origSFT = rdd.first.getType
-    val builder = new SimpleFeatureTypeBuilder
-    builder.setName(typeName)
-    builder.addAll(origSFT.getAttributeDescriptors)
-    val sft = builder.buildFeatureType
+  def normalizeShpRdd(rdd: RDD[SimpleFeature], typeName: String)(implicit sc: SparkContext) =
+    rdd.mapPartitions({ featureIter =>
+      val bufferedFeatures = featureIter.buffered
+      val headFeature = bufferedFeatures.head
 
-    rdd.map(convertToSFT(sft))
-  }
+      val builder = new SimpleFeatureTypeBuilder
+      builder.addAll(headFeature.getType.getAttributeDescriptors)
+      builder.setName(typeName)
```

Comment:
I think the user data is on the SFT. If you're building a new one, shouldn't you need to copy it over from old to new? Could do a quick test on the REPL...

-----Original Message-----
From: "Nathan Zimmerman" <notifications@github.com>
Sent: ‎9/‎8/‎2016 4:10 PM
To: "azavea/geowave-geomesa-comparative-analysis" <geowave-geomesa-comparative-analysis@noreply.github.com>
Cc: "jpolchlo" <jpolchlopek@azavea.com>; "Comment" <comment@noreply.github.com>
Subject: Re: [azavea/geowave-geomesa-comparative-analysis] Fix geotoolsretyping for shapefile ingest (#15)

In empirical-data/common/src/main/scala/com/azavea/ingest/common/HydrateRDD.scala:
>  
> -    rdd.map(convertToSFT(sft))
> -  }
> +      val builder = new SimpleFeatureTypeBuilder
> +      builder.addAll(headFeature.getType.getAttributeDescriptors)
> +      builder.setName(typeName)
This should just change the type - if there's user data on these features, I think it should remain. Perhaps there's a gap in my reasoning here
—
You are receiving this because you commented.
Reply to this email directly, view it on GitHub, or mute the thread.